spec/services/cap_quantity_spec.rb

```diff
@@ -83,5 +83,38 @@
         end
       end
     end
+
+    context "when no items are available" do
+      it "sets quantity to 0 for unavailable items, and reports the change" do
+        changes = {}
+
+        expect {
+          changes = CapQuantity.new.call(order)
+          [line_item1, line_item2, line_item3].each(&:reload)
+        }.to change { line_item1.quantity }.to(0)
+          .and change { line_item2.quantity }.to(0)
+          .and change { line_item3.quantity }.to(0)
+
+        expect(changes[line_item1.id]).to eq 3
+        expect(changes[line_item2.id]).to eq 3
+        expect(changes[line_item3.id]).to eq 3
```

Comment:
I don't understand the purpose of the `changes` attribute. I would expect the "changes" to be `-3` in this case. 
Rather, it seems to be the original quantity of the line item before the cap was applied, which is quite a different meaning (and I'm not sure is useful at all).

🤷 I can see that this is already how the CapQuantity class works, so I guess it's all good.